modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache

```diff
@@ -397,7 +397,7 @@ namespace {{packageName}}.{{apiPackage}}
                     {{^required}}
 
                     if ({{paramName}}.IsSet)
-                        uriBuilderLocalVar.Path = uriBuilderLocalVar.Path + $"/{ Uri.EscapeDataString({{paramName}}.Value).ToString()) }";
+                        uriBuilderLocalVar.Path = uriBuilderLocalVar.Path + $"/{ Uri.EscapeDataString({{paramName}}.Value).ToString() }";
```

Comment:
this part or code block is for optional path parameters.

from what I know, path parameters should be required all the time.

can you share the use cases in which path parameter is optional?

cc @devhl-labs

Reply:
That appears to be correct. Looks like the {{^required}} section can be removed.

https://swagger.io/docs/specification/describing-parameters/#path-parameters:~:text=Also%20remember%20to%20add%C2%A0required%3A%20true%2C%20because%20path%20parameters%20are%20always%20required.

Reply:
Yeah this is right, we actually had an path param that was marked as optional(not on purpose, was supposed to be a query param, fixed already).
I didn't really mean to change any logic to the generation, just the parenthesis that was causing a build error there if it fell into that case.

Reply:
👌 

I've filed https://github.com/OpenAPITools/openapi-generator/pull/17525 instead to remove optional path parameters so closing this one.